### PR TITLE
JWNG-555 - finish form documentation

### PR DESCRIPTION
**Pull request by mignonat**

### CODE REVIEW

File: public/core.d.ts [hunk 1]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
    * Here you'll find all form related methods
    */
   namespace Form {
+
+    /**
+     * ***JMap.Form.getFormsMetaDataByLayerId***
+     * 
+     * Returns all forms metadata for a given layer id.
+     * 
+     * Fecth data from server first time, then keeps it in cache for the next time.
```
Comment (jean-kiepura) [inline public/core.d.ts:8817]: **_Fecthes_** data from server _**the**_ first time, then keeps it in cache for the next time.

File: public/core.d.ts [hunk 2]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
     function getFormsMetaDataByLayerId(layerId: JId): Promise<JFormMetaData[]>
+
+    /**
+     * ***JMap.Form.getElement***
+     * 
+     * Returns form data of an element from given parameters.
+     * 
+     * Works only for attribute forms.
+     * 
+     * @param params params which identifies the element
```
Comment (jean-kiepura) [inline public/core.d.ts:8839]: _**identify**_ (3ième personne du pluriel)

File: public/core.d.ts [hunk 3]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
     function getElement(params: JFormElementId): Promise<JFormElement | undefined>
+
+    /**
+     * ***JMap.Form.getElements***
+     * 
+     * Returns form data of multiple elements from given parameters.
+     * 
+     * Works only for attribute forms.
+     * 
+     * @param params params which identifies the elements
```
Comment (jean-kiepura) [inline public/core.d.ts:8863]: **_identify_** (3ième personne du pluriel)

File: public/core.d.ts [hunk 4]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
     function getElements(params: JFormElementIds): Promise<JFormElement[]>
+
+    /**
+     * ***JMap.Form.getEntries***
+     * 
+     * Returns form data of multiple entries from given parameters.
+     * 
+     * Works only for external and sub forms.
+     * 
+     * @param params params which identifies the entries
```
Comment (jean-kiepura) [inline public/core.d.ts:8887]: idem identify

File: public/core.d.ts [hunk 5]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *  const formsMetadata = JMap.Form.getAllFormsMetaDataForCurrentLayer()
+     * }
+     * ```
+     */
     function getAllFormsMetaDataForCurrentLayer(): JFormMetaData[]
+
+    /**
+     * ***JMap.Form.getAllFormsMetaDataForCurrentLayer***
+     * 
+     * Returns form metadata of the current displayed layer and given form id.
```
Comment (jean-kiepura) [inline public/core.d.ts:9212]: pourquoi "current displayed" ?  C'est quoi le contexte? "displayed" où?

(on devrait dire "currently displayed" aussi, comme plus bas)

File: public/core.d.ts [hunk 6]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *  JMap.Form.hasAttributeForm()
+     * }
+     * ```
+     */
     function hasAttributeForm(): boolean
+
+    /**
+     * ***JMap.Form.getAttributeForm***
+     * 
+     * Returns attribute form of the current displayed layer.
```
Comment (jean-kiepura) [inline public/core.d.ts:9243]: _**currently**_ displayed

File: public/core.d.ts [hunk 7]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     * 
+     * @throws if layer has no attribute form
+     * @example ```ts
+     * 
+     *  // returns attribute form of the current displayed layer
```
Comment (jean-kiepura) [inline public/core.d.ts:9248]: idem currently

File: public/core.d.ts [hunk 8]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *  const attributeForm = JMap.Form.getAttributeForm()
+     * }
+     * ```
+     */
     function getAttributeForm(): JForm
+
+    /**
+     * ***JMap.Form.getExternalForms***
+     * 
+     * Returns external forms of the current displayed layer.
```
Comment (jean-kiepura) [inline public/core.d.ts:9258]: currently

File: public/core.d.ts [hunk 9]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     * 
+     * @return an empty array if layer has no external forms
+     * @example ```ts
+     * 
+     *  // returns external forms of the current displayed layer
```
Comment (jean-kiepura) [inline public/core.d.ts:9263]: currently

File: public/core.d.ts [hunk 10]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *  const externalForms = JMap.Form.getExternalForms()
+     * }
+     * ```
+     */
     function getExternalForms(): JForm[]
+
+    /**
+     * ***JMap.Form.getSubForms***
+     * 
+     * Returns sub forms of the current displayed layer.
```
Comment (jean-kiepura) [inline public/core.d.ts:9273]: currently (2 fois)

File: public/core.d.ts [hunk 11]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *      ]
+     *    )
+     *    .then(formMetadata => console.log("Update sub form dialog opened", formMetadata))
+     *    .catch(error => console.error(error))
+     * ```
+     */
     function openUpdateDialogForSubForm(subFormId: JId, subFormElements: JFormElement[]): JFormMetaData
+
+    /**
+     * ***JMap.Form.closeCurrentDisplayedDialog***
```
Comment (jean-kiepura) [inline public/core.d.ts:9406]: hum...... ici je pense qu'on aurait du nommer la fonction "JMap.Form.close**_Currently_**DisplayedDialog" mais bon ça fait du refacto qui risque d'affecter des extensions ....

File: public/core.d.ts [hunk 12]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *    type: "tree"
+     *  })
+     * ```
+     */
     function setFormValues(form: JForm, attributeValueByName: JAttributeValueByName): JFormErrors
+
+    /**
+     * ***JMap.Form.submit***
+     * 
+     * Set form values of the given form.
```
Comment (jean-kiepura) [inline public/core.d.ts:9456]: Mauvaise description? 

"submits the form to the server"?

File: public/core.d.ts [hunk 13]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     * // returns true if the current form attribute elements can be deleted
+     * JMap.Form.canDeleteCurrentElements()
+     * ```
+     */
     function canDeleteCurrentElements(): boolean
+
+    /**
+     * ***JMap.Form.deleteCurrentElements***
+     * 
+     * Delete the current attribute form elements.
```
Comment (jean-kiepura) [inline public/core.d.ts:9504]: Delete**_s_**

File: public/core.d.ts [hunk 14]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     * ```
+     */
     function getNextViewId(): number
+
+    /**
+     * ***JMap.Form.incrementNextViewId***
+     * 
+     * Increments the next view id.
+     * 
+     * This is a technical method that you should never used.
```
Comment (jean-kiepura) [inline public/core.d.ts:9538]: "that you should never have to use"

File: public/core.d.ts [hunk 15]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     * // increments the next view id
+     * JMap.Form.incrementNextViewId()
+     * ```
+     */
     function incrementNextViewId(): void
+    
+    /**
+     * ***JMap.Form.processRule***
+     * 
+     * Processes a JSON Logic rule and returns the result, undefined if rules is not correct.
```
Comment (jean-kiepura) [inline public/core.d.ts:9551]: ", or undefined if ...."

File: public/core.d.ts [hunk 16]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *  ...
+     * }
+     * ```
+     */
     function getDisplayedFormPhotos(): JPhoto[]
+    
+    /**
+     * ***JMap.Form.addDisplayedFormPhoto***
+     * 
+     * Add a photo to the currently displayed form.
```
Comment (jean-kiepura) [inline public/core.d.ts:9682]: "Adds"

File: public/core.d.ts [hunk 17]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *  ...
+     * }
+     * ```
+     */
     function addDisplayedFormPhoto(photo: JPhoto): JId
+    
+    /**
+     * ***JMap.Form.updateDisplayedFormPhoto***
+     * 
+     * Update a photo to the currently displayed form.
```
Comment (jean-kiepura) [inline public/core.d.ts:9709]: "Updates"

File: public/core.d.ts [hunk 18]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     *  }
+     * }
+     * ```
+     */
     function updateDisplayedFormPhoto(params: JFormPhotoUpdate): void
+    
+    /**
+     * ***JMap.Form.removeDisplayedFormPhotoById***
+     * 
+     * Delete a photo on the currently displayed form, for a given photo id.
```
Comment (jean-kiepura) [inline public/core.d.ts:9733]: "Deletes"

File: public/core.d.ts [hunk 19]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     * ```
+     */
     function closeCurrentDisplayedDialog(): void
+
+    /**
+     * ***JMap.Form.getFormValues***
+     * 
+     * Returns default values if form has not been edited, else form values.
+     * 
+     * This is a technical method used by JMap App, you should never used this function
```
Comment (TomAvediss) [inline public/core.d.ts:9425]: you should never used this function => you should never have to use this function ou you should never use this function

File: public/core.d.ts [hunk 20]
```diff
@@ -8808,48 +8808,942 @@ declare namespace JMap {
...
+     * ```
+     */
     function deleteCurrentElements(): Promise<JFormDeleteResult>
+
+    /**
+     * ***JMap.Form.getNextViewId***
+     * 
+     * Returns the next view id.
+     * 
+     * This is a technical method that you should never used.
```
Comment (TomAvediss) [inline public/core.d.ts:9523]: that you should never used => that you should never use / that you should never have to use (comme Jean a dit plus bas)

**Review by jean-kiepura — APPROVED** [review]

**Pull request merged**